src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobieUserRepositoryInterpreter.scala

```diff
@@ -48,24 +47,26 @@ class DoobieUserRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
 
   import UserSQL._
 
-  def create(user: User): F[User] =
+  def put(user: User): F[User] =
     insert(user).withUniqueGeneratedKeys[Long]("ID").map(id => user.copy(id = id.some)).transact(xa)
 
-  def update(user: User): F[Option[User]] = OptionT.fromOption[ConnectionIO](user.id).semiflatMap { id =>
-    UserSQL.update(user, id).run
-  }.value.as(user.some).transact(xa)
+  def updateSafe(user: User): OptionT[F, User] = OptionT.fromOption[F](user.id).flatMapF { id =>
+    UserSQL.update(user, id).run.as(user.some).transact(xa)
+  }
 
-  def get(userId: Long): F[Option[User]] = select(userId).option.transact(xa)
+  def update(user: User): F[User] = OptionT.fromOption[F](user.id).semiflatMap { id =>
+    UserSQL.update(user, id).run.transact(xa)
+  }.value.as(user)
 
-  def findByUserName(userName: String): F[Option[User]] =
-    byUserName(userName).option.transact(xa)
+  def get(userId: Long): OptionT[F, User] = OptionT(select(userId).option.transact(xa))
```

Comment:
I know this is from tsec BackingStore, but I like the idea of OptionT.  I will create an issue we can track later to move all `F[Option[A]]` to `OptionT[F, A]` in the rest of the code

Reply:
Reverted for now, I made the mistake of basing this branch off of the branch for PR #79. I like your issue #82 idea.